locations/models.py

```diff
@@ -192,4 +195,5 @@ def get_context(self, request):
         # Update template context
         context = super().get_context(request)
         context['locationpages'] = locationpages
+        context['parentpage'] = LocationsIndexPage.objects.live().parent_of(self).first()
```

Comment:
move this up to where locationpages is initialized or remove locationpages variable and do `LocationPage.objects.filter(tags__name=tag).live().descendant_of(self.get_parent())` in line